doc/extending.md

```diff
@@ -61,3 +61,30 @@ can run the templating manually:
 ```
 ./baselayer/tools/fill_conf_values.py --config="config.yaml" static/js/component.jsx.template
 ```
+
+## Adding external services
+
+External services are microservices that you can pull from GitHub and run as part of your application. This is useful for integrating third-party services or custom scripts.
+
+If you want to add external services to your application,
+you can do so by adding them to the `config.yaml` file under the `services.external` key. This allows baselayer to pull the external service from a GitHub repository.
+
+The configuration in the `config.yaml` file should look like this:
+
+```
+services:
+  external:
+    my_service:
+        url: "https://github.com/my_service.git"
+        branch: main
+        sha: specific_commit_sha
+        params: {}
+    another_service:
+        url: "https://github.com/another_service.git"
+        version: v0.1.0
+        params: {}
+```
+
+You must provide the `url` of the GitHub repository. To target a specific version of the service, you can optionally include both the `branch` and `sha`, or use the `version` field to refer to a particular release. Additional configuration options can be passed through the `params` dictionary, which the service may use during setup.
+
+The external service will then be initialized and registered in supervisor, alongside other services, provided it is correctly configured and compatible with the application.
```

Comment:
See for example here you talk about "compatibility" but it's never defined before, hence my previous comment